misk-admin/src/main/kotlin/misk/web/metadata/config/ConfigMetadataAction.kt

```diff
@@ -47,18 +48,19 @@ class ConfigMetadataAction @Inject constructor(
     config: Config
   ): Map<String, String?> {
     val rawYamlFiles = MiskConfig.loadConfigYamlMap(appName, deployment, listOf())
-    val yamlFiles = linkedMapOf<String, String?>()
+    val configFileContents = linkedMapOf<String, String?>()
     when (mode) {
       ConfigTabMode.SAFE -> {
-        yamlFiles.put("JVM", jvmMetadataAction.getRuntime())
+        configFileContents.put("Effective Config", MiskConfig.toRedactedYaml(config, ResourceLoader.SYSTEM))
+        configFileContents.put("JVM", jvmMetadataAction.getRuntime())
       }
       ConfigTabMode.UNSAFE_LEAK_MISK_SECRETS -> {
-        yamlFiles.put("Effective Config", MiskConfig.toYaml(config, ResourceLoader.SYSTEM))
-        rawYamlFiles.map { yamlFiles.put(it.key, it.value) }
-        yamlFiles.put("JVM", jvmMetadataAction.getRuntime())
+        configFileContents.put("Effective Config", MiskConfig.toRedactedYaml(config, ResourceLoader.SYSTEM))
```

Comment:
i would expect the unsafe `toYaml` rather than `toRedactedYaml` here?

Reply:
I renamed the `toYaml` function to `toRedactedYaml`

Reply:
I think we should simplify the safe/unsafe logic. Something like:

```
  private fun generateConfigResources(
    appName: String,
    deployment: Deployment,
    config: Config
  ): Map<String, String?> {
    val rawYamlFiles = MiskConfig.loadConfigYamlMap(appName, deployment, listOf())
    val configFileContents = linkedMapOf<String, String?>()

    configFileContents.put("Effective Config", MiskConfig.toRedactedYaml(config, ResourceLoader.SYSTEM))
    if (mode == ConfigTabMode.UNSAFE_LEAK_MISK_SECRETS) {
      rawYamlFiles.forEach { configFileContents.put(it.key, it.value) }
    }

    configFileContents.put("JVM", jvmMetadataAction.getRuntime())
    return configFileContents
  }
```